### PR TITLE
MLW-1005: "Eligibility" alerts should drive queue eligibility

### PR DESCRIPTION
**Pull request by rhenshaw56**

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -268,7 +269,13 @@ const utils = {
     let retValue = false;
 
     for (let i = 0; i < alerts.length; i++) {
-      retValue =  alerts[i].categories.some(a => category.indexOf(a) >= 0);
+      let alertCategories = alerts[i].categories;
+      retValue =  alertCategories.every(
+        a => categories.every(
```
Comment (jomadoye) [inline src/utils.js]: @rhenshaw56 I haven't worked on the alert code before, so I don't fully understand this PR, but please the "a" and "b" variable names to something more descriptive.

File: src/constants.js [hunk 2]
```diff
@@ -633,14 +633,37 @@ export const VIRAL_LOAD_ALERTS = [
 ];
 
 export const VIRAL_LOAD_ALERTS_CATEGORIES = [
```
Comment (mogoodrich) [inline src/constants.js]: @rhenshaw56  you followed the pattern that already existed, which made sense, but taking a closer look I think we should change all these arrays into single category constants, ie:

`VIRAL_LOAD_ALERTS_CATEGORY = "viral-load"`

and then add a SCREENING_ELIGIBILITY_CATEGORY.  (There may be a time when we want to, for example, search for an alert that *just* has the blood pressure category).

Then all the filters would be updated to something like this:

`return utils.hasAlert(patient.alert, [ADHERENCE_ALERTS_CATEGORY, SCREENING_ELIGIBILITY_CATEGORY])`

if you wanted we could make a single const like:

```
const ALERT_CATEGORIES = {
    VIRAL_LOAD: "vl",
    HTC: "HTC"
  ...
}
```

Comment (rhenshaw56) [reply src/constants.js]: @mogoodrich I have made the changes the you requested

Comment (cioan) [reply src/constants.js]: Thanks @rhenshaw56 and @mogoodrich , this looks good!

**Pull request merged**